src/core/wappu-mood-core.js

```diff
@@ -0,0 +1,152 @@
+const {knex} = require('../util/database').connect();
+import {deepChangeKeyCase} from '../util';
+import * as feedCore from './feed-core';
+const requireEnvs = require('../util/require-envs');
+
+requireEnvs(['MOOD_START_DATE', 'MOOD_END_DATE']);
+
+function createOrUpdateMood(opts) {
+  const upsertMoodSql = `
+    WITH upsert AS
+      (UPDATE
+        wappu_mood
+      SET
+        rating = ?,
+        description = ?,
+        updated_at = CURRENT_TIMESTAMP
+      WHERE
+        user_id = ? AND created_at_coarse = CURRENT_DATE
+      RETURNING
+        *,
+        false AS is_new
+    ), inserted AS (
+        INSERT INTO
+          wappu_mood(user_id, rating, description)
+        SELECT ?, ?, ? WHERE NOT EXISTS( SELECT * FROM upsert )
+        RETURNING
+          *,
+          true AS is_new
+    )
+    SELECT *
+    FROM upsert
+    UNION ALL
+    SELECT *
+    FROM inserted;
+  `;
+
+  let params = [
+    opts.rating, opts.description, opts.client.id,
+    opts.client.id, opts.rating, opts.description
+  ];
+
+  return knex.transaction(trx =>
+    trx.raw(upsertMoodSql, params)
+      .then(result => {
+        const mood = result.rows[0];
+        if (mood.is_new) feedCore.createFeedItem(_feedTemplate(mood, opts.client));
+        return undefined;
+      })
+      .catch(err => {
+        err.message = 'Error updating database';
+        err.status = 500;
+        throw err;
+      })
```

Comment:
There should be no need for catch here as we have an error middleware to take care of these. It also takes care of doing proper logging when the exception is unexpected and makes sure the error is not leaked to the client if the error is unexpected.

See `src/middlewares/error-logger.js` and `src/middlewares/error-responder.js`